app/controllers/users_controller.rb

```diff
@@ -815,6 +815,12 @@ def preferences
     render json: { message: 'ok' }, status: :ok
   end
 
+  def get_preferences
+    raise Exceptions::UnprocessableEntity, 'No current user!' if !current_user
+    user = User.find(current_user.id)
+    render json: { message: 'ok', header_pre:  user.preferences[:header_pre]}, status: :ok
```

Comment:
This controller should not return only the one preference but all that are available. This limits the scope unnecessarily.